File: CampingNeretva/UI/campingneretva_mobile/lib/models/user_model.dart
import 'package:campingneretva_mobile/models/user_type_model.dart';

class User {
  final int id;
  final String username;
  final String email;
  final String firstName;
  final String lastName;
  final UserType userType;

  User({
    required this.id,
    required this.username,
    required this.email,
    required this.firstName,
    required this.lastName,
    required this.userType,
  });

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      id: json['userId'],
      username: json['userName'],
      email: json['email'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      userType: UserType.fromJson(json['userType']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userName': username,
      'email': email,
      'firstName': firstName,
      'lastName': lastName,
      'userTypeId': userType.id,
    };
  }
}

File: CampingNeretva/CampingNeretva.Model/Models/UserModel.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace CampingNeretva.Model.Models
{
    public class UserModel
    {
        public int UserId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string UserName { get; set; }
        public string Email { get; set; }
        public string? PhoneNumber { get; set; }
        public UserTypeModel UserType { get; set; }
    }
}
